kaamaurice/OpenPype
# Add subset lock system for blender

This MR adds a system that adds lock subset in the db system.
Also adds working behaviour in Blender.
Here are some details:

Lock is removed when Blender is closed (using atexit lib)
Lock is invalid when flush delay is passed (defined in project's settings, necessary if blender crashes and can't unlock)
Lock is invalid when lock site is the same as user's local site

Implementation:

All the functional code was added to a lock.py file in openpype/pipeline.
An operator and a timer were added to the ops.py file in hosts/blender/api.

How to test:

Open a blender file from OpenPype
Once the scene is opened, close blender
Go onto another computer (see bellow if you can't switch computer) and open the same blender file from OpenPype
A popup should indicate you that the file is locked and that you can QUIT or PROCEED anyway

If you can't switch computer:
As the site is the same between the lock one and yours, the only solution is to edit hosts/blender/api/lock.py locked_on_same_site() and compare with anything but get_local_site_id().
